src/functionalTest/java/uk/gov/hmcts/reform/sendletter/config/MockConfiguration.java

```diff
@@ -1,21 +1,30 @@
 package uk.gov.hmcts.reform.sendletter.config;
 
 import org.mockito.Mockito;
-import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationContext;
 import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Primary;
 import uk.gov.hmcts.reform.sendletter.cache.SentLettersCache;
-import uk.gov.hmcts.reform.sendletter.cache.SentLettersInMemoryCache;
 import uk.gov.hmcts.reform.sendletter.services.LetterChecksumGenerator;
 
-@TestConfiguration
+@Configuration
 public class MockConfiguration {
 
-    public static final LetterChecksumGenerator GENERATOR = Mockito.mock(LetterChecksumGenerator.class);
+    @Autowired
+    private ApplicationContext context;
 
     @Bean
     @Primary
-    public SentLettersCache withGeneratorMock() {
-        return new SentLettersInMemoryCache(GENERATOR);
+    public SentLettersCache withGeneratorMock(@Value("${redis.enabled}") boolean isRedisEnabled) {
```

Comment:
can this be renamed to `sentLetterCache` now?

Reply:
can rename - no biggy